rotkehlchen/globaldb/updates/parsers.py

```diff
@@ -0,0 +1,334 @@
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BaseParser(abc.ABC, Generic[T]):
+    """Base parser with common parsing functionality.
+
+    This was introduced to handle asset updates that need to be applied
+    before global db schema version changes. By having dedicated parser classes for different
+    data types (assets, collections, mappings), we can ensure compatible updates
+    are applied before the schema transitions to a new version.
+    """
+    def __init__(self) -> None:
+        self._double_quotes_re = re.compile(r'\"(.*?)\"')
+        self._string_re = re.compile(r'.*([\'"])(.*?)\1.*')
+        self._version_parsers: list[tuple[VersionRange, Callable]] = []
+
+    def parse(self, insert_text: str, version: int, connection: 'DBConnection') -> T:
+        for version_range, parser in self._version_parsers:
+            if version_range.contains(version):
+                return parser(insert_text=insert_text, connection=connection)
+
+        raise ValueError(f'No parser found for version {version}')
+
+    def _parse_value(self, value: str) -> str | int | None:
+        match = self._string_re.match(value)
+        if match is not None:
+            return match.group(2)
+
+        value = value.strip()
+        if value == 'NULL':
+            return None
+
+        try:
+            return int(value)
+        except ValueError:
+            return value
+
+    def _parse_str(self, value: str, name: str, insert_text: str) -> str:
+        result = self._parse_value(value)
+        if not isinstance(result, str):
+            raise DeserializationError(
+                f'Got invalid {name} {value} from {insert_text}',
+            )
+        return result
+
+    def _parse_optional_str(self, value: str, name: str, insert_text: str) -> str | None:
+        result = self._parse_value(value)
+        if result is not None and not isinstance(result, str):
+            raise DeserializationError(
+                f'Got invalid {name} {value} from {insert_text}',
+            )
+        return result
+
+    def _parse_optional_int(self, value: str, name: str, insert_text: str) -> int | None:
+        result = self._parse_value(value)
+        if result is not None and not isinstance(result, int):
+            raise DeserializationError(
+                f'Got invalid {name} {value} from {insert_text}',
+            )
+        return result
+
+    def standardize_quotes(self, text: str) -> str:
+        """Convert double quotes to single quotes for consistency.
+
+        We enforce single quotes for all SQL values. If any old updates
+        come with double quotes, we convert them here.
+        """
+        def replace_double_quotes(match_result: re.Match[str]) -> str:
+            content = match_result.group(1).replace("'", "''")
+            return f"'{content}'"
+
+        return self._double_quotes_re.sub(replace_double_quotes, text)
+
+
+class AssetParser(BaseParser[AssetData]):
```

Comment:
small docstring here.

Also question: Shouldn't this specify which versions its' for? I guess for all known versions until now? At least in the docstring specify it.